### DISCUSSION TITLE
Suggestion: Keep verbs with verbs

### DISCUSSION THREAD

**Discussion by scorster** [musescore/MuseScore#16224] [Ideas]
I find it odd that MuseScore 4's **Add Linked Staff** "verb" is buried with various Staff Property "adjectives."

Why not reposition the **Add Linked Staff** "verb" directly along side the **Add Staff** verb? As shown here and the bottom of the image:

     ![Add Linked STaff next to Add Staff](https://user-images.githubusercontent.com/121324720/216891643-bc93487f-ce24-46c1-b008-e3084319552c.png)

But MuseScore would need to toggle **Add Linked staff** between enabled and disabled:

• MuseScore 4 would enable Add Linked Staff only when the user has a single staff selected. 
• Accordingly, when the user has an entire Instrument selected MuseScore would disable the Add Linked Staff button. 

This matches MS 3.6's behavior, so the recommend change not a stretch logically or programatically.

Here's a specific example regarding the posted image above:

• if the user has the **Acoustic Guitar** Instrument selected MuseScore disables the **Add Linked Staff** button.

• when the user selects the **Treble Clef 8va staff bass** staff (a child of **Acoustic Guitar**) the **Add Linked Staff** button is enabled.

**Comment by cbjeukendrup** [comment]
"Add linked staff" applies to each staff of the instrument, while "Add staff" is just one option for the whole instrument. This might not be relevant for instruments that only have a single staff, but Piano and Organ have multiple staves of course. When an instrument has multiple staves, you need a way to choose for which of them you want to create a linked staff. 

(That's not to say that I don't think the current situation is not ideal; it's just to explain why it is as it is currently, and why your suggestion is a bit like "too good to be true". 🙂)

**Reply by scorster** [reply]
Glad you see some merit in my suggestion, but I'm slightly perplexed your reply. Wondering if you missed the latter half of my original post, below the image. For clarification I've expanded it some since the initial post.

My suggestion is directly parallel to the behavior in MS3's Instrument panel.

In the MS3's Instrument panel the Add Linked Staff button is enabled only when Staff:1 or Staff:2 ... is selected. The button is disabled when an Instrument is selected. Is there something that prevents MuseScore 4 from operating that way?

**Reply by cbjeukendrup** [reply]
@scorster Apologies. I completely missed the text below the image, probably because I was working on a small screen 🤦 

Now that I've read the whole story: that makes sense to me! I'll let the design team know about it.

**Comment by ShibuyaCyana** [comment]
Agree, too many clicks needed to add a linked tablature staff for a guitar in MS4, while in MS3 I could do this when creating a score...
![screenshot](https://github.com/musescore/MuseScore/assets/39703247/8f271abb-95a3-468c-9223-f9e8ffb51641)
In MS3 new score wizard:
`select staff - add linked staff - click dropdown - select tablature`, done;
In MS4:
`expand instrument - click staff gear icon - add linked staff - click new staff gear icon - click dropdown - select tablature`

Looks like only 2 clicks away, but I can't help to highlight the staff before clicking on the gear icon, fearing of changing properties for other staffs, which adds another 2 or 3 clicks (including instrument); and then another click to un-expand the instrument to make it looks, um, untouched? I don't know if this only happens to me...

Other possible improvements:
1. make guitar (staff + tab) as an instrument, since we already have two separate "instruments" for guitar and guitar (tablature);
2. Automatically create linked staff as tablature, if instrument applies. Maybe even let user choose which type by default, possibly from the "advanced style properties" dialog.
![dialog](https://github.com/musescore/MuseScore/assets/39703247/99d1d2cc-1a84-480b-af3b-245a2344d941)

**Comment by scorster** [comment]
Ultimately, it will be music welcomed when MuseScore has a one-stop shopping **Properties>Guitar** panel that houses all relevant Staff settings, Staff>Advanced Style Properties, tuning data ... 

And a **New Guitar Score wizard** that initially walks one through all the settings for a guitar instrument, including the option of linked staves.

**Comment by scorster** [comment]
Issue persists in 4.5.2, amidst a slightly more cluttered environment * **

![Add Linked Staff next to Add Staff 4-5-2](https://github.com/user-attachments/assets/f21432a9-6b7f-4e7c-a8dd-d0f647290229)

The comment at the bottom of the panel: 

     Note : linked staves contain identical information.

is struggling to say something while failing to convey an important exception.

     • The pitch between linked staves is identical
     • The notation can be quite different as with any of the non-standard TAB staff types.

* The panel now sports a non-operational **System markings** stub (that's installed and slated for some future functionality?)
** The Add button/menu also includes a **System markings** ... option, which appears to be disabled under all circumstances.

We're these aspects meant to be included in the 4.5.2 release?
